src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -308,6 +319,7 @@ public PPLTool create(Map<String, Object> map) {
                 (String) map.getOrDefault("prompt", ""),
                 (String) map.getOrDefault("model_type", ""),
                 (String) map.getOrDefault("previous_tool_name", ""),
+                Integer.valueOf((String) map.getOrDefault("head", "-1")),
```

Comment:
Can we make this a non-optional field or give it a default value, when it's -1 and the response is long, this is still possible to exceed the max tokens to llm.

Reply:
I prefer to let customer to define this. The head number should be related to the data inside index. If the data like log, is very long, the head number should be small. If we want to set a default number for that, it should be a larger one without breaking some correct answer. So I prefer a large number like 200 or just leave as optional one.